types/index.d.ts

```diff
@@ -0,0 +1,434 @@
+
+    // TODO
+    indoors: any;
+    themes: any;
+    routes: any;
+    buildings: any;
+    props: any;
+    indoorMapEntities: any;
+    indoorMapFloorOutlines: any;
+    heatmaps: any;
+}
+
+/* L.Marker */
+
+type MarkerOptions = L.MarkerOptions & {
+    elevation: number;
+    elevationMode: ElevationMode;
+    indoorMapId: IndoorMapId;
+    indoorMapFloorId: IndoorMapFloorId;
+};
+
+class Marker extends L.Marker {
+    constructor(latlng: L.LatLngExpression, options?: MarkerOptions);
+    getElevation(): number;
+    setElevation(elevation : number): this;
+    getElevationMode(): this;
+    setElevationMode(elevationMode: ElevationMode): this;
+    setIndoorMapWithFloorId(indoorMapId: IndoorMapId, indoorMapFloorId: IndoorMapFloorId): this;
+    setIndoorMapWithFloorIndex(indoorMapId: IndoorMapId, indoorMapFloorIndex: IndoorMapFloorIndex): this;
+    setOutdoor(): this;
+}
+
+/* L.Popup */
+
+type PopupOptions = L.PopupOptions & {
+    elevation: number;
+};
+
+class Popup extends L.Popup {
+    constructor(options?: PopupOptions, source?: L.Layer);
+    getElevation(): number;
+    setElevation(elevation : number): this;
+}
+
+/* Wrld.Polygon */
+
+type PolygonOptions = {
+    color?: Color;
+    elevation?: number;
+    elevationMode?: ElevationMode;
+    indoorMapId?: IndoorMapId;
+    indoorMapFloorId?: IndoorMapFloorId;
+};
+
+class Polygon {
+    constructor(latlngs: L.LatLngTuple[] | L.LatLngTuple[][], options?: PolygonOptions);
+    addTo(map: Map): this;
+    remove(): this;
+    getColor(): Color;
+    setColor(color: Color): this;
+    getPoints(): L.LatLngLiteral[];
+    addHole(points: L.LatLngTuple[]): this;
+    getHoles(): L.LatLngLiteral[][];
+}
+
+/* Wrld.Polyline */
+
+type PolylineOptions = {
+    color?: Color;
+    elevation?: number;
+    elevationMode?: ElevationMode;
+    indoorMapId?: IndoorMapId;
+    indoorMapFloorId?: IndoorMapFloorId;
+    weight?: number;
+    miterLimit?: number;
+};
+
+class Polyline extends L.Polyline {
+    constructor(latlngs: L.LatLngExpression[], options?: PolylineOptions);
+    getPoints(): L.LatLng[];
+    getIndoorMapId(): IndoorMapId;
+    getIndoorMapFloorId(): IndoorMapFloorId;
+    setIndoorMapWithFloorId(indoorMapId: IndoorMapId, indoorMapFloorId: IndoorMapFloorId): this;
+    getElevation(): number;
+    setElevation(elevation : number): this;
+    getElevationMode(): this;
+    setElevationMode(elevationMode: ElevationMode): this;
+    getWidth(): number;
+    getColor(): Color;
+    getMiterLimit(): number;
+    setOptions(options: PolylineOptions): this;
+    setStyle(options: PolylineOptions): this;
+}
+
+/* Wrld.Prop */
+
+type PropOptions = {
+    elevation?: number;
+    elevationMode?: ElevationMode;
+    indoorMapId?: IndoorMapId;
+    indoorMapFloorId?: IndoorMapFloorId;
+    headingDegrees?: number;
+};
+
+class Prop {
+    constructor(name: string, geometryId: string, location: L.LatLngExpression, options?: PropOptions);
```

Comment:
Similar to getAltitudeAtLatLng() - our code assumes we can address a lat and lng property of the object so we can't take a LatLngTuple

Reply:
This is already wrapped by L.latLng https://github.com/wrld3d/wrld.js/blob/types/src/public/prop.js#L8